### ISSUE TITLE
Reduce usage of xdmp:eval in bootstrap

### ISSUE THREAD

**Issue by grtjn** [marklogic/roxy#782]
The bootstrap code in setup.xqy relies a lot on xdmp:eval, more than necessary. Several cleanup rounds have been executed already, but this is to go over a last time to see if there is anything else that could be improved in this area.

Thoughts:

- use xdmp:functions (available since ML6) to build up a map:map of available functions, or use fn:function-available (available down to ML5). Use xdmp:function and xdmp:apply to invoke if available (both available down to ML5).
- replace some of the evals for accessing security data with at least sem functions for that purpose, or if possible and available xdmp functions that don't need eval against Security..

**Comment by RobertSzkutak** [comment]
I'm thinking we should remove any code in place to support versions < ML7

**Comment by grtjn** [comment]
Yeah, me too. Was thinking of taking multiple steps at first, but then i noticed rest-extension-user being used since 1.7.1. And i als believe map:contains is not supported in ml6. It wouldn't be difficult to bypass, but since nobody complained it seems there is no longer need..

Do you know how long ago ml6 became obsolete?

**Comment by RobertSzkutak** [comment]
@grtjn June 2016
